sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1457,8 +1458,29 @@ class DatasetSuite extends QueryTest
       }
 
       testCheckpointing("basic") {
-        val ds = spark.range(10).repartition($"id" % 2).filter($"id" > 5).orderBy($"id".desc)
-        val cp = if (reliable) ds.checkpoint(eager) else ds.localCheckpoint(eager)
+        val ds = spark
+          .range(10)
+          // Num partitions is set to 1 to avoid a RangePartitioner in the orderBy below
+          .repartition(1, $"id" % 2)
+          .filter($"id" > 5)
+          .orderBy($"id".desc)
+        @volatile var jobCounter = 0
+        spark.sparkContext.addSparkListener(new SparkListener {
```

Comment:
we should remove this listener at the end of test.

Reply:
Good call. I will use the withListener helper.